File: head_to_offer/09/hto_09.py
class CQueue:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def appendTail(self, value: int) -> None:
        self.stack1.append(value)

    def deleteHead(self) -> int:
        if not self.stack2:
            if not self.stack1:
                return -1
            else:
                while (self.stack1):
                    self.stack2.append(self.stack1.pop())
        return self.stack2.pop()


if __name__ == "__main__":
    queue = CQueue()
    queue.appendTail(1)
    print(queue.deleteHead())
    print(queue.deleteHead())

File: head_to_offer/09/hto_09.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.Deque;
import java.util.ArrayDeque;
import java.util.List;
import java.util.ArrayList;

class CQueue {
    Deque<Integer> stack1;
    Deque<Integer> stack2;

    public CQueue() {
        stack1 = new ArrayDeque<>();
        stack2 = new ArrayDeque<>();
    }
    
    public void appendTail(int value) {
        stack1.add(value);
    }
    
    public int deleteHead() {
        if (stack2.isEmpty()) {
            if (stack1.isEmpty()) return -1;
            while (!stack1.isEmpty())
                stack2.add(stack1.pop());
        }
        return stack2.pop();
    }
}

public class leetcode {
    public static void main(String[] args) {
        CQueue queue = new CQueue();
        queue.appendTail(1);
        System.out.println(queue.deleteHead());
        System.out.println(queue.deleteHead());
    }
}